### PR TITLE
Update to rust stable and futures 0.3.

### PR DESCRIPTION
**Pull request by orium**
Land #52 first, since there is some intersection.

### CODE REVIEW

**Comment by orium** [comment]
Since there is no rust-highfive for this project and it is kinda of a low activity repo I'm going to mention @cramertj to ask for a review :slightly_smiling_face:

File: examples/06_03_select/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#![allow(unused)]
```
Comment (cramertj) [inline examples/06_03_select/src/lib.rs]: why is this needed? can it be removed?

File: src/04_pinning/01_chapter.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ works. Consider the following code:
 ```rust
 let fut_one = ...;
 let fut_two = ...;
-async move {
+async {
```
Comment (cramertj) [inline src/04_pinning/01_chapter.md]: why was this changed? the previous syntax was as-intended.

**Comment by cramertj** [comment]
Thanks for the PR! two nits but then I'm happy to merge.

File: src/04_pinning/01_chapter.md (hunk 2 above)
Comment (orium) [reply src/04_pinning/01_chapter.md]: my bad.  reverting it

**Comment by cramertj** [comment]
I've merged #52 which I believe obsoletes this. Feel free to reopen if that's incorrect.
